DataAccessLayer/Repositories/Interfaces/ICrudRepository.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DataAccessLayer.Repositories.Interfaces
+{
+    interface ICrudRepository<T> where T : class
+        {
+            IEnumerable<T> GetAll();
```

Comment:
As written above, this pattern is limiting you. For example, it does not cover situations of:
- getting paginated portions of data
- get fitered data
- bulk add/edit/delete
- and more specific cases that may happen in future projects